### ISSUE TITLE
Working with web.api

### ISSUE THREAD

**Issue by matejgolob** [object/Simple.OData.Client#88]
I have begginer question. Is there any "working" example how to "correctly" build server endpoint with asp.net web.api, and "working" client with simple.odata?

I've created simple server endpoint like example:
http://www.c-sharpcorner.com/UploadFile/4b0136/sacffolding-the-web-api-2-odata-controller-in-visual-studio/
I can access odata with browser address like  "http://localhost:2693/api/employee"

After that I've created new console application, installed simple.odata with nuget, wrote 
            var client = new ODataClient("http://localhost:2693/api/");
            var employees = await client
                .For<Employee>()
                .FindEntriesAsync();
but I get error:
System.Web.Http.HttpResponseException occurred
Message: A first chance exception of type 'System.Web.Http.HttpResponseException' occurred in System.Web.Http.dll
Additional information:
 Processing of the HTTP request resulted in an exception. Please see the HTTP response returned by the 'Response' property of 
this exception for details.

Thanks for help.

Matej

**Comment by object** [comment]
Thank you for the issue report. I will build this example and check what's going on.

**Comment by object** [comment]
Hi,
I followed the steps in the article and everything worked like a charm. The only difference was that I had to use /odata suffix instead of /api, but I suppose this is because of different VS 2013 minor version. AFAIK the library is often used in combination with Web API 2, so there must be something with your setup. Can you give me more details, e.g. Fiddler HTTP communication logs? Or share your project?

**Comment by matejgolob** [comment]
Thanks for reply. I've done it again. simple solution with two projects - server + console client. Now I dont' get any exception but still doesn't work... Probably  I'm doing some basic mistake...
Here is zipped solution: https://app.box.com/s/y0ejn1rp5n2p9cpphlmw0gftseau9h05
Thanks for help in advice.

**Comment by object** [comment]
You need to call Test1().Wait() in your Main method, not just Test1(). And change Test1 method signature from void to Task. Then it should work.

**Comment by matejgolob** [comment]
It "should" but It doesn't... Now I get:
An unhandled exception of type 'System.AggregateException' occurred in mscorlib.dll
inside "Wait".

Here is callstack:

   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at SimpleOdataTest.Program.Main(String[] args) in c:\Users\matejg\Documents\Visual Studio 2013\Projects\SimpleODataTest\SimpleOdataTest\Program.cs:line 14
   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
   at System.AppDomain.ExecuteAssembly(String assemblyFile, Evidence assemblySecurity, String[] args)
   at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.ThreadHelper.ThreadStart()

**Comment by object** [comment]
Actually I changed for my tests the NuGet package to use Simple.OData.Client, not Simple.OData.V4.Client. And it worked. And when I changed it back to V4 client it didn't. Of course, your Web API server is V3 server, check its metadata: edmx:DataServices m:DataServiceVersion="3.0" m:MaxDataServiceVersion="3.0. And you are using V4 client that simple can't understand it.

You have two options:

- Use Simple.OData.V3.Client
- Use Simple.OData.Client that includes them both. It loads twice as many files but it transparently selects the right adapter version.
